File: calc.py
def calculate(first_number, operator, second_number):
	"""Return the result of applying an arithmetic operator."""
	if operator in ("+", "add"):
		return first_number + second_number
	if operator in ("-", "subtract"):
		return first_number - second_number
	if operator in ("*", "multiply"):
		return first_number * second_number
	if operator in ("/", "divide"):
		if second_number == 0:
			raise ValueError("cannot divide by zero")
		return first_number / second_number
	raise ValueError("operator must be +, -, *, or /")


def main():
	try:
		first_number = float(input("First number: "))
		operator = input("Operator (+, -, *, /): ").strip().lower()
		second_number = float(input("Second number: "))
		result = calculate(first_number, operator, second_number)
	except ValueError as error:
		print(f"Error: {error}")
		return

	print(f"Result: {result:g}")


if __name__ == "__main__":
	main()

File: calc.java
import java.util.Scanner;

public class calc {
	static double calculate(double firstNumber, String operator, double secondNumber) {
		switch (operator) {
			case "+":
			case "add":
				return firstNumber + secondNumber;
			case "-":
			case "subtract":
				return firstNumber - secondNumber;
			case "*":
			case "multiply":
				return firstNumber * secondNumber;
			case "/":
			case "divide":
				if (secondNumber == 0) {
					throw new IllegalArgumentException("cannot divide by zero");
				}
				return firstNumber / secondNumber;
			default:
				throw new IllegalArgumentException("operator must be +, -, *, or /");
		}
	}

	public static void main(String[] args) {
		Scanner scanner = new Scanner(System.in);

		try {
			System.out.print("First number: ");
			double firstNumber = scanner.nextDouble();
			scanner.nextLine();

			System.out.print("Operator (+, -, *, /): ");
			String operator = scanner.nextLine().trim().toLowerCase();

			System.out.print("Second number: ");
			double secondNumber = scanner.nextDouble();

			double result = calculate(firstNumber, operator, secondNumber);
			System.out.println("Result: " + result);
		} catch (java.util.InputMismatchException error) {
			System.out.println("Error: numbers must be valid numeric values");
		} catch (IllegalArgumentException error) {
			System.out.println("Error: " + error.getMessage());
		} finally {
			scanner.close();
		}
	}
}
